### PR TITLE
Preparation to refactor Suse::Backend

### PR DESCRIPTION
**Pull request by mdeniz**
This PR is preparing the actual Suse::Backend to be refactored in the future. The idea behind that refactoring is to be able to have classes and methods that knows how to set the urls and what http methods to use instead of having to know it from every place we are using the backend.

The number of files changed is "big" but, the steps done are simple:

* Suse::Backend got renamed as Backend::Connection and moved to a new directory lib/backend, also now is autoloaded instead of having the need to require it
* Moved the model BackendFile and renamed as Backend::File, that way is part of the same namespace
* Fix some rubocop offenses on the way because some files were touched while renaming the Suse::Backend calls

### CODE REVIEW

**Comment by ChrisBr** [comment]
LGTM
Btw: Are we actually using the BackendFile`s at the moment?

**Comment by mdeniz** [comment]
Yes @ChrisBr , we use it! See:

```ruby
class PackageFile < BackendFile
class PackageServiceErrorFile < PackageFile

#lines 894-897 of Package model
def service_error(revision = nil)
  revision ||= serviceinfo.try{ to_hash['xsrcmd5'] }
  PackageServiceErrorFile.new(project_name: project.name, package_name: name).to_s(rev: revision)
end
```

And also:

```ruby
class ProjectFile < BackendFile
class ProjectConfigFile < ProjectFile
#lines 164-166 of Project model
def config
  @config ||= ProjectConfigFile.new(project_name: name)
end

#line 536 of SourceController
config = project.is_a?(String) ? ProjectConfigFile.new(project_name: project) : project.config

```

**Comment by evanrolfe** [comment]
LGTM

File: src/api/config/application.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Application < Rails::Application
 
     # Custom directories with classes and modules you want to be autoloadable.
     # config.autoload_paths += %W(#{config.root}/extras)
+    config.eager_load_paths << Rails.root.join('lib')
```
Comment (hennevogel) [inline src/api/config/application.rb]: Uhm there is so much more stuff in lib. Have you checked that this is okay for everything?

Comment (mdeniz) [inline src/api/config/application.rb]: No, but I  can also restrict it to just the backend code... let me try.

**Pull request merged**